src/hooks/useSwappableUserAssets.ts

```diff
@@ -0,0 +1,126 @@
+import { SwappableAsset } from '@/entities';
+import { walletFilter } from '@/handlers/tokenSearch';
+import { Network } from '@/helpers';
+import { useAssetsInWallet, useCoinListEditOptions } from '@/hooks';
+import { ETH_ADDRESS } from '@/references';
+import {
+  EthereumAddress,
+  ETH_ADDRESS as ETH_ADDRESS_AGGREGATORS,
+} from '@rainbow-me/swaps';
+import { ethereumUtils } from '@/utils';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+type SwappableAddresses = {
+  [Network.mainnet]: EthereumAddress[];
+  [Network.optimism]: EthereumAddress[];
+  [Network.polygon]: EthereumAddress[];
+  [Network.arbitrum]: EthereumAddress[];
+};
+
+export const useSwappableUserAssets = (params: {
+  outputCurrency: SwappableAsset;
+}) => {
+  const { outputCurrency } = params;
+  const assetsInWallet = useAssetsInWallet() as SwappableAsset[];
+  const { hiddenCoinsObj } = useCoinListEditOptions();
+  const [swappableAssets, setSwappableAssets] = useState<SwappableAddresses>({
+    [Network.mainnet]: [],
+    [Network.optimism]: [],
+    [Network.polygon]: [],
+    [Network.arbitrum]: [],
+  });
+
+  const filteredAssetsInWallet = useMemo(
+    () => assetsInWallet.filter(asset => !hiddenCoinsObj[asset.uniqueId]),
+    [assetsInWallet, hiddenCoinsObj]
+  );
+
+  const getSwappableAddressesForNetwork = useCallback(
+    async (addresses: EthereumAddress[], network: Network) => {
```

Comment:
i think `addresses` is not used

Reply:
ahh, I was generating an address list in two place. Probably a remnant of my original implementation prior to refactor. This value is now being used.